File: Source/Fantasy.Base/XSerialization/XDateTimeConverter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel;
using System.Globalization;

namespace Fantasy.XSerialization
{
    [XConverter(TargetType = typeof(DateTime))]
    public class XDateTimeConverter : TypeConverter
    {
        public override object ConvertFrom(ITypeDescriptorContext context, System.Globalization.CultureInfo culture, object value)
        {

            return ((DateTime)value).ToString("O");
        }

        public override object ConvertTo(ITypeDescriptorContext context, System.Globalization.CultureInfo culture, object value, Type destinationType)
        {
            DateTime rs = DateTime.MinValue;
            string text = (string)value;
            if (!string.IsNullOrEmpty(text))
            {
                if(!DateTime.TryParseExact(text, "O", null, DateTimeStyles.None, out rs))
                {
                    rs = DateTime.Parse(text);
                }
                
            }
            return rs;
        }
    }
}

File: branches/Stheno/AutoConverted/Fantasy.Base/XSerialization/XDateTimeConverter.java
package Fantasy.XSerialization;

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//[XConverter(TargetType = typeof(java.util.Date))]
public class XDateTimeConverter extends TypeConverter
{
	@Override
	public Object ConvertFrom(ITypeDescriptorContext context, System.Globalization.CultureInfo culture, Object value)
	{

		return ((java.util.Date)value).ToString("O");
	}

	@Override
	public Object ConvertTo(ITypeDescriptorContext context, System.Globalization.CultureInfo culture, Object value, java.lang.Class destinationType)
	{
		java.util.Date rs = java.util.Date.getMinValue();
		String text = (String)value;
		if (!DotNetToJavaStringHelper.isNullOrEmpty(text))
		{
			RefObject<java.util.Date> tempRef_rs = new RefObject<java.util.Date>(rs);
			boolean tempVar = !java.util.Date.TryParseExact(text, "O", null, DateTimeStyles.None, tempRef_rs);
				rs = tempRef_rs.argvalue;
			if (tempVar)
			{
				rs = new java.util.Date(java.util.Date.parse(text));
			}

		}
		return rs;
	}
}